primitives/arithmetic/src/fixed.rs

```diff
@@ -0,0 +1,1311 @@
+	+ Saturating + Bounded
+	+ Eq + PartialEq + Ord + PartialOrd
+	+ CheckedSub + CheckedAdd + CheckedMul + CheckedDiv
+	+ Add + Sub + Div + Mul
+{
+	/// The underlying data type used for this fixed point number.
+	type Inner: Copy + Debug;
+
+	/// The accuracy of this fixed point number.
+	const DIV: Self::Inner;
+
+	/// Accuracy of this `Fixed` implementation.
+	fn accuracy() -> Self::Inner {
+		Self::DIV
+	}
+
+	/// Raw constructor. Assigns `int / Self::accuracy()` to `self`.
+	fn from_inner(int: Self::Inner) -> Self;
+
+	/// Consumes `self` and returns the inner raw value (`self * Self::accuracy()`).
+	fn into_inner(self) -> Self::Inner;
+
+	/// Creates self from an integer number `int`.
+	/// 
+	/// Returns `Self::max` or `Self::min` if `int` exceeds accuracy.
+	fn from_integer<N: UniqueSaturatedInto<Self::Inner>>(int: N) -> Self;
+
+	/// Creates `self` from an integer number `int`.
+	/// 
+	/// Returns `None` if `int` exceeds accuracy.
+	fn checked_from_integer(int: Self::Inner) -> Option<Self>;
+
+	/// Creates `self` from a rational number. Equal to `n / d`.
+	///
+	/// Panics if `d = 0`. Returns `Self::max` or `Self::min` if `n / d` exceeds accuracy.
+	fn from_rational<N: FixedPointOperand, D: FixedPointOperand>(n: N, d: D) -> Self;
+
+	/// Creates `self` from a rational number. Equal to `n / d`.
+	///
+	/// Returns `None` if `d == 0` or `n / d` exceeds accuracy.
+	fn checked_from_rational<N: FixedPointOperand, D: FixedPointOperand>(n: N, d: D) -> Option<Self>;
+
+	/// Checked multiplication for integer type `N`. Equal to `self * n`.
+	///
+	/// Returns `None` if the result does not fit in `N`.
+	fn checked_mul_int<N: FixedPointOperand>(self, n: N) -> Option<N>;
+
+	/// Saturating multiplication for integer type `N`. Equal to `self * n`.
+	///
+	/// Returns `N::min` or `N::max` if the result does not fit in `N`.
+	fn saturating_mul_int<N: FixedPointOperand>(self, n: N) -> N;
+
+	/// Checked division for integer type `N`. Equal to `self / d`.
+	///
+	/// Returns `None` if the result does not fit in `N` or `d == 0`.
+	fn checked_div_int<N: FixedPointOperand>(self, d: N) -> Option<N>;
+
+	/// Saturating division for integer type `N`. Equal to `self / d`.
+	///
+	/// Panics if `d == 0`. Returns `N::min` or `N::max` if the result does not fit in `N`.
+	fn saturating_div_int<N: FixedPointOperand>(self, d: N) -> N;
+
+	/// Saturating multiplication for integer type `N`, adding the result back.
+	/// Equal to `self * n + n`.
+	///
+	/// Returns `N::min` or `N::max` if the multiplication or final result do not fit in `N`.
+	fn saturating_mul_int_acc<N: FixedPointOperand>(self, n: N) -> N;
+
+	/// Saturating absolute value.
+	///
+	/// Returns `Self::max` if `self == Self::min`.
+	fn saturating_abs(self) -> Self;
+
+	/// Takes the reciprocal (inverse). Equal to `1 / self`.
+	///
+	/// Returns `None` if `self = 0`.
+	fn reciprocal(self) -> Option<Self> {
+		Self::one().checked_div(&self)
+	}
+
+	/// Returns zero.
+	fn zero() -> Self;
```

Comment:
such functions should be implemented on the trait, no? if possible this would be good. Then you need to import this trait in less places.

Reply:
Yeah, I moved them.